src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -2521,7 +2530,7 @@ private function registerHttpClientConfiguration(array $config, ContainerBuilder
         if ($responseFactoryId = $config['mock_response_factory'] ?? null) {
             $container->register('http_client.mock_client', MockHttpClient::class)
                 ->setDecoratedService('http_client.transport', null, -10)  // lower priority than TraceableHttpClient (5)
-                ->setArguments([new Reference($responseFactoryId)]);
+                ->setArguments(true === $responseFactoryId ? [] : [new Reference($responseFactoryId)]);
```

Comment:
What is the goal here ? Tell me if I'm wrong but it looks like you are trying to set it to `[]` only if at least one dedicated mock was set. But then what if you have a global one and one specific for only one of the clients ? 

Because that variable is set only during foreach.

Reply:
"Because that variable " <-- do you mean $responseFactoryId which is set in the if ?

Reply:
yup It is created and overidden in foreach at each iteration. So the only value it could have would be `the last scoped client says its true`

Reply:
I'll look into this :)